src/control_node_startup_script.sh

```diff
@@ -42,40 +57,40 @@ sudo chown -R 1000:100 /home/ec2-user/.aws/
 
 # Deploy cluster cloudformation stack. This includes the EKS, EFS, Autoscaler, and Loadbalancer
 # This script needs output from the control node cloudformation stack
-python3 deploy_cluster_cf.py --control-node-stackname ${STACK_NAME}
+python3 deploy_cluster_cf.py --control-node-stackname "$STACK_NAME"
 
 # Wait for the cluster cloudformation stack to complete before continuing...
-aws cloudformation wait stack-create-complete --stack-name "umsi-easy-hub-${TAG}-cluster"
+aws cloudformation wait stack-create-complete --stack-name "umsi-easy-hub-$TAG-cluster"
 
 # Get output of cloudformation stack
-output=($(python3 get_cluster_cf_output.py --cluster-stackname "umsi-easy-hub-${TAG}-cluster") )
+IFS=" " read -r -a output <<< "$(python3 get_cluster_cf_output.py --cluster-stackname "umsi-easy-hub-$TAG-cluster")"
+echo "${output[*]}"
+export EKS_NAME="${output[0]}"
+export NODE_ROLE_ARN="${output[1]}"
+export ASG_ARN="${output[2]}"
 # ${output[0]} = Tag
 # ${output[1]} = EksName
 # ${output[2]} = NodeRoleArn
 # ${output[3]} = Asg
 
-# Get kubectl binary which will expose control plane configuration options
-curl -o kubectl https://amazon-eks.s3-us-west-2.amazonaws.com/1.11.5/2018-12-06/bin/linux/amd64/kubectl
-chmod +x ./kubectl
-sudo cp ./kubectl /usr/local/bin/kubectl
-
-# Get aws-iam-authenticator to authenticate kubectl binary with our EKS backplane
-curl -o aws-iam-authenticator https://amazon-eks.s3-us-west-2.amazonaws.com/1.11.5/2018-12-06/bin/linux/amd64/aws-iam-authenticator
-chmod +x ./aws-iam-authenticator
-sudo cp ./aws-iam-authenticator /usr/local/bin/aws-iam-authenticator
-
-# Install aws cli
-# echo yes | sudo amazon-linux-extras install python3
-sudo rm /usr/bin/aws
-pip3 install --upgrade awscli --user
-sudo cp ~/.local/bin/aws /usr/bin/aws
-
 # Sync kubectl with the EKS we want
-aws eks update-kubeconfig --name ${output[1]}
+aws eks update-kubeconfig --name "$EKS_NAME"
 curl -O https://amazon-eks.s3-us-west-2.amazonaws.com/cloudformation/2019-01-09/aws-auth-cm.yaml
-sed -i -e "s;<ARN of instance role (not instance profile)>;${output[2]};g" aws-auth-cm.yaml
+sed -i -e "s;<ARN of instance role (not instance profile)>;$NODE_ROLE_ARN;g" aws-auth-cm.yaml
 kubectl apply -f aws-auth-cm.yaml
 
+# Upgrade some internal components of the cluster
+eksctl utils update-kube-proxy --cluster "$EKS_NAME" --approve
+eksctl utils update-aws-node --cluster "$EKS_NAME" --approve
+eksctl utils update-coredns --cluster "$EKS_NAME" --approve
```

Comment:
Do you really need to update these components before installing the addon?
Since you are installing the last Kube (1.19, right?), I would expect everything to be up to date...